agent/inbound.go

```diff
@@ -55,13 +55,36 @@ func (a *Agent) processIncomingApplication(ev *event.Event) error {
 		return err
 	}
 
+	sourceUIDMatch, err := a.appManager.CompareSourceUID(a.context, incomingApp)
+	if err != nil {
+		return fmt.Errorf("failed to compare the source UID of app: %w", err)
+	}
+
 	switch ev.Type() {
 	case event.Create:
+		if !sourceUIDMatch {
+			logCtx.Debug("An app already exists with a different source UID. Deleting the existing app")
+			if err := a.deleteApplication(incomingApp); err != nil {
+				return err
```

Comment:
I think we should give more context to the returned error here.

The caller will expect an application to be created, and probably reflect that in any error message logged, but could potentially receive an error from the delete operation. 

For example, the following case would be confusing (admittedly, the chance for this happening is very slim, but it should ):

1. A create event is sent for an application where `sourceUIDMatch` is false
2. The application is deleted on the cluster by a third party
3. `a.deleteApplication` fails because the resource already is deleted. The error would be "the requested resource could not be found"
4. That error is passed back to the caller
5. The error log for the caller would look similar to "Could not create app foo: requested resource not found"

Or think about RBAC - agent might have RBAC to create applications, but not delete applications, but the create call would return a permission denied. Troubleshooting that would be fun :)

So it would make more sense to augment the returned error by more information, e.g. `fmt.Errorf("could not delete existing resource prior to creation: %w", err)` or something similar.

Reply:
Makes sense. Added more context to errors.